physical/consul/consul.go

```diff
@@ -240,65 +238,80 @@ func NewConsulBackend(conf map[string]string, logger log.Logger) (physical.Backe
 	return c, nil
 }
 
-func setupTLSConfig(conf map[string]string) (*tls.Config, error) {
-	serverName, _, err := net.SplitHostPort(conf["address"])
-	switch {
-	case err == nil:
-	case strings.Contains(err.Error(), "missing port"):
-		serverName = conf["address"]
-	default:
-		return nil, err
+func setupTLSConfig(conf map[string]string, tlsConfig *api.TLSConfig, logger log.Logger) (*tls.Config, error) {
+	address, ok := conf["address"]
+	if ok {
+		tlsConfig.Address = address
 	}
 
-	insecureSkipVerify := false
-	if _, ok := conf["tls_skip_verify"]; ok {
-		insecureSkipVerify = true
+	serverName, ok := conf["tls_server_name"]
```

Comment:
The default behavior still needs to be the net.SplitHostPort behavior, if not overridden with tls_server_name.

Reply:
`SetupTLSConfig` extracts the host portion from the `TLSConfig.Address` using `net.SplitHostPort` which is why I did not repeat the effort here.

```go
	if tlsConfig.Address != "" {
		server := tlsConfig.Address
		hasPort := strings.LastIndex(server, ":") > strings.LastIndex(server, "]")
		if hasPort {
			var err error
			server, _, err = net.SplitHostPort(server)
			if err != nil {
				return nil, err
			}
		}
		tlsClientConfig.ServerName = server
	}
```